nuxeo-core/nuxeo-core-management/src/main/java/org/nuxeo/ecm/core/management/probes/ProbeManagerImpl.java

```diff
@@ -218,4 +231,61 @@ protected void doRunProbe(ProbeInfo probe) {
         }
     }
 
+    @Override
+    public void registerProbeForHealthCheck(HealthCheckProbesDescriptor descriptor) {
+        String name = descriptor.getName();
+        if (!descriptor.isEnabled()) {
+            if (probesForHealthCheck.containsKey(name)) {
+                probesForHealthCheck.remove(name);
+                return;
+            }
+        }
+        if (infosByShortcuts.containsKey(name)) {
+            probesForHealthCheck.put(name, getProbeInfo(name));
+        }
+    }
+
+    @Override
+    public Collection<ProbeInfo> getHealthCheckProbes() {
+        return Collections.unmodifiableCollection(probesForHealthCheck.values());
+    }
+
+    @Override
+    public HealthCheckResult getOrRunHealthChecks() {
+        int refreshSeconds = Integer.parseInt(Framework.getProperty(DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS_PROPERTY,
```

Comment:
Should be initialized at the start of the component or extracted into a method as it is used twice (see `getOrRunHealthChecek(String name)`